scripts/utils/dictionary.txt

```diff
@@ -0,0 +1,723 @@
+aspecto
```

Comment:
this file doesn't make sense to me. E.g. why are commit hashes in there, or things like p50 or src? If the checker is so dumb as to flag every single string, then there's no value in it for us, it's always going to be too noisy.

Reply:
Hmm, i get your point. Could you recommend a local spellchecker that runs well? I'll use that instead

Reply:
I cannot, research is part of the task. I think OTEL website used one, other CNCF websites like Kubernetes might use one too.

Reply:
So I researched a few spelling checker libraries and found 2 potential candidates. One is GNU Aspell and the other is [misspell](https://pkg.go.dev/github.com/client9/misspell/cmd/misspell) by client9. I tested both of them and I found Aspell to perform better at detecting wrong spellings and ignoring false positives like hexadecimal strings. This leaves us with 4 options (if we include the old options as well): Aspell, misspell, cspell (GitHub action), and pyspellchecker (in decreasing order of preference). I can continue looking for more spellcheckers if these are not satisfactory.

Reply:
I wrote this implementation in Aspell. I created a dictionary on the basis of the initial run of Aspell over the contents. I tried my best to not accidentally add actual typos or misspellings generated from the command while writing the dictionary. One problem I encountered was of encoding of words like `Kröhling` and `Paixão`. Aspell was struggling to deal with the unicode characters of these words despite setting the encoding to UTF-8. The alternative was to use Hunspell, but it performs worse overall when it comes to checking spellings. Apart from that, the spellcheck seems to be working fine.

Reply:
> words like Kröhling and Paixão

does it not work if we add these words to the dictionary?

Reply:
On running the script I get the following error (after adding these words to the dictionary) 
![image](https://github.com/jaegertracing/documentation/assets/34743284/63f3d332-5e83-4078-b5f4-0f8f088949fa)